src/concurrency/concurrency_handler.rs

```diff
@@ -0,0 +1,51 @@
+use std::rc::Rc;
+
+use super::data_race;
+use crate::concurrency::GenmcCtx;
+use crate::{VisitProvenance, VisitWith};
+
+pub enum ConcurrencyHandler {
+    None,
+    DataRace(Box<data_race::GlobalState>),
+    GenMC(Rc<GenmcCtx>),
```

Comment:
Following the usual Rust naming convention, this should be `GenMc`. Or maybe `Genmc`, that's what you picked for `GenmcCtx`. Either way, the variant name and the `*Ctx` type name should be consistent.